server/pkg/store/astra/db.go

```diff
@@ -308,3 +310,291 @@ func (a *AstraServerStore) getClusterID(orgID, clusterName string) (string, erro
+		return nil, fmt.Errorf("failed to get icon: %w", err)
+	}
+	cqlDescription, err := client.ToString(result.Rows[0].Values[12])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get launch ui redirect url: %w", err)
+	}
+	cqlLaunchUiValues, err := client.ToString(result.Rows[0].Values[13])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get launch ui values: %w", err)
+	}
+	cqlOverrideValues, err := client.ToString(result.Rows[0].Values[14])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get override values: %w", err)
+	}
+	cqlReleaseNameValues, err := client.ToString(result.Rows[0].Values[15])
+	if err != nil {
+		return nil, fmt.Errorf("failed to get override values: %w", err)
+	}
+
+	config := &types.AppConfig{
+		Name:                cqlAppName,
+		ChartName:           cqlChartName,
+		RepoName:            cqlRepoName,
+		RepoURL:             cqlRepoURL,
+		Namespace:           cqlNamespace,
+		Version:             cqlVersion,
+		CreateNamespace:     cqlCreateNamespace,
+		PrivilegedNamespace: cqlPrivilegedNamespace,
+		LaunchUIURL:         cqlLaunchUiUrl,
+		LaunchUIRedirectURL: cqlLaunchUiRedirectUrl,
+		Category:            cqlCategory,
+		Icon:                cqlIcon,
+		Description:         cqlDescription,
+		LaunchUIValues:      cqlLaunchUiValues,
+		OverrideValues:      cqlOverrideValues,
+		ReleaseName:         cqlReleaseNameValues,
+	}
+
+	return config, nil
+}
+
+func (a *AstraServerStore) GetAppsFromStore() (*[]types.AppConfig, error) {
+
+	selectQuery := &pb.Query{
+		Cql: fmt.Sprintf("Select name,chart_name,repo_name,repo_url,namespace,version,create_namespace,privileged_namespace,launch_ui_url,launch_ui_redirect_url,category,icon,description,launch_ui_values,override_values, release_name FROM %s.app_config;",
+			a.keyspace),
+	}
+
+	response, err := a.c.Session().ExecuteQuery(selectQuery)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialise db: %w", err)
+	}
+
+	result := response.GetResultSet()
+
+	if len(result.Rows) == 0 {
+		return nil, fmt.Errorf("app config's not found")
+	}
+
+	var appConfigs []types.AppConfig
+	for _, row := range result.Rows {
+		cqlAppName, err := client.ToString(row.Values[0])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get app name: %w", err)
+		}
+		cqlChartName, err := client.ToString(row.Values[1])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get chart name: %w", err)
+		}
+		cqlRepoName, err := client.ToString(row.Values[2])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get repo name: %w", err)
+		}
+		cqlRepoURL, err := client.ToString(row.Values[3])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get repo url: %w", err)
+		}
+		cqlNamespace, err := client.ToString(row.Values[4])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get Namespace: %w", err)
+		}
+		cqlVersion, err := client.ToString(row.Values[5])
+		if err != nil {
+			return nil, fmt.Errorf("failed to get version: %w", err)
+		}
```

Comment:
reading DB values can be optimized for both gep app methods with common method convert resultset to struct